### PR TITLE
SQL-1017: Add formula for supporting endswith

### PR DESCRIPTION
**Pull request by nbagnard**

### CODE REVIEW

**Review by bucaojit — APPROVED** [review]
LGTM!

File: connector/dialect.tdd [hunk 1]
```diff
@@ -1054,6 +1054,11 @@
      <argument type='str' />
      <argument type='str' />
     </function>
+    <function group='string' name='ENDSWITH' return-type='bool'>
+      <formula>(CASE WHEN (CHAR_LENGTH(%1) - CHAR_LENGTH(%2)) &lt; 0 THEN false ELSE SUBSTRING(%1 FROM (CHAR_LENGTH(%1) - CHAR_LENGTH(%2)) FOR (CHAR_LENGTH(%2))) = %2 END)</formula>
```
Comment (mattChiaravalloti) [inline connector/dialect.tdd]:
```suggestion
      <formula>(CASE WHEN CHAR_LENGTH(%1) &lt; CHAR_LENGTH(%2) THEN false ELSE SUBSTRING(%1 FROM (CHAR_LENGTH(%1) - CHAR_LENGTH(%2)) FOR (CHAR_LENGTH(%2))) = %2 END)</formula>
```
why not do `char_length(%1) < char_length(%2)` and save us the additional operation?

**Review by mattChiaravalloti — APPROVED** [review]
looks good but I think it can be better -- no need to retag me, I'm ok with this staying the same or changing (the difference is likely very small)

File: connector/dialect.tdd (hunk 1 above)
Comment (nbagnard) [reply connector/dialect.tdd]: Seems very reasonable to me 👍

**Comment by nbagnard** [comment]
evergreen merge

**Pull request merged**